storage/dataflux/next_page.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataflux
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	// wsDefaultPageSize specifies the number of object results to include in a single page for worksteal listing.
+	wsDefaultPageSize = 1000
+)
+
+// nextPageOpts specifies options for next page of listing result .
+type nextPageOpts struct {
+	// startRange is the start offset of the objects to be listed.
+	startRange string
+	// endRange is the end offset of the objects to be listed.
+	endRange string
+	// bucketHandle is the bucket handle of the bucket from which objects are to be listed.
+	bucketHandle *storage.BucketHandle
+	// query is the storage.Query to filter objects for listing.
+	query storage.Query
+	// skipDirectoryObjects is to indicate whether to skip or list directory objects.
+	skipDirectoryObjects bool
+	// generation is the generation number of the last object in the page.
+	generation int64
+}
+
+// nextPageResult represents the results of fetching a single page of objects
+// from a GCS listing operation and information for remaining objects to be listed.
+type nextPageResult struct {
+	// items is the list of objects listed.
+	items []*storage.ObjectAttrs
+	// doneListing indicates whether the lister has completed listing.
+	doneListing bool
+	// nextStartRange is the start offset of the next page of objects to be listed.
+	nextStartRange string
+	// generation is the generation number of the last object in the page.
```

Comment:
This is the generation number of the last object in the previous page, right?

Reply:
nextPageResult stores result of the current page listed.
items are the objects fetched from gcs. We list 1000 objects and discard the last one. The generation number of the last object(discarded) is used for next call.